### ISSUE TITLE
Disable "Screen-to-Sleep" & Enable an "Always-On" mode for the screen

### ISSUE THREAD

**Issue by TOGLK** [stream-pi/client#45]
What Platform is targeted?:
 Android
 
Is this a Bug-Fix, Feature Request, or, Other?:
 Feature Request

Describe the issue:
 Currently, when using the Stream-Pi Client Application on an Android device the screen is able to enter a "Sleep" state 
 when it is not used for a short period of time. This behaviour is the standard behaviour of Android when no element has 
 been instated to stop the device from entering a "Sleep" state, and so because the screen is able to power off after 5- 
 Seconds (Default "Screen-Off" time) their is now way to use Stream-Pi for more than between 5 & 15 seconds after the 
 screen of the Android device was last interacted with.
 
What resolution do you suggest?:
 Implement the relevant permissions and application changes to allow for the Android App to override the standard screen 
 "Wake-Lock TImeout", thus allowing the screen to remain on and Stream-Pi being free to use all of the time from your 
 chosen Android device.
 
What other things should be considered?:
  With this change it is important that a WARNING is provided when the Android App is first launched stating that use of the 
  App on devices with a OLED, AMOLED, SAMOLED, or, S-SAMOLED can result in what is commonly known as "Burn-In", 
  Which is when the image shown ont he display is brunt into the display itself leaving a permanent degradation on the 
  screen which you cannot repair despite many App's that may claim otherwise.
  This occurs because OLED displays are organic and the organic elements that are used degrade over time naturally, 
  static images cause the screen to degrade more on those parts of the display that are brighter than the rest.

**Comment by rnayabed** [comment]
There is NO way to detect what TYPE of screen the phone has. The only method is to query device details, and then search it up on a list whether it has an OLED (or any such screens with burn-in risks) display or not. 

Or, I can also provide a disclaimer that turning on the "do not sleep" toggle may cause burn in if the device is left ON for a lot of time and if the device has an OLED screen

**Comment by TOGLK** [comment]
> There is NO way to detect what TYPE of screen the phone has. The only method is to query device details, and then search it up on a list whether it has an OLED (or any such screens with burn-in risks) display or not.
> 
> Or, I can also provide a disclaimer that turning on the "do not sleep" toggle may cause burn in if the device is left ON for a lot of time and if the device has an OLED screen

My current view on this is to provide an application-wide warning that is shown on ALL supported "Client Platforms".

The warning would state text similar to the following & would be shown at start-up on the first use, and, on the tenth use of the application, where on the 10th use of the "Stream-Pi Client" the text would update slightly to say "Reminder" instead of "WARNING!".

This warning would only appear when the "Do Not Sleep" toggle that you have suggested above is switched to the "ON"/"Activated" position, either set of wording could be used.

Burn-In CAN effect ANY screen type, but, on other screen types the effects can technically be reversed, on OLED displays the effect CANNOT be reversed, this is on the contrary to quite a few YouTube videos and written text "Guide's" on the internet that claim to have a "Magical" way to "FIX" your burn-in on the OLED display.

OLED displays and any display using the same organic materials in their display technology are prone to Burn-In, it is a deficiency of the technology and cannot be overcome, it is for this reason that manufacturers are already working on alternative display technology to replace OLED in every use-case.

So, in short, we don't need to detect what "TYPE" of screen the device has, as we can simply just enable to "WARNING" that is displayed when the toggle is activated by the end-user.

**Comment by rnayabed** [comment]
an OLED Burn In option has been provided in settings
